File: web/filing/models/income/salary/salary-model.ts
import { EmployerAddressModel } from "./employer-address-model"
import { EmploymentPeriodModel } from "./employment-period-model"
import { SalarySection172Model } from "./salary-section-172-model"
import { EmployerModel } from "./employer-model"
import { SalarySection171Model } from "./salary-section-171-model"
import { SalaryDeduction16Model } from "./salary-deduction-16-model"
import { SalarySection173Model } from "./salary-section-173-model"

export interface SalaryModel {
    employer: EmployerModel
    employerAddress: EmployerAddressModel
    employmentPeriod: EmploymentPeriodModel
    salaryDeduction16: SalaryDeduction16Model
    salarySection171: SalarySection171Model[]
    salarySection172: SalarySection172Model[]
    salarySection173: SalarySection173Model[]
  }

  // Props for SalaryIncomeWidget component
  export interface SalaryIncomeWidgetProps {
    isOpen: boolean
    onClose: () => void
    filingId: number | null
    entityId?: number | string
    initialData?: Record<string, any> | null
    onSuccess?: (action: 'created' | 'updated', data: any) => void
    errorField?: string | null
    errorMessage?: string | null  // Error message to display for the field
    fieldErrors?: Record<string, string> | null  // Multiple field errors: { fieldName: errorMessage }
  }
  

  
  export interface SalaryMasterData {
    states?: Array<{ value: string; label: string }>
    countries?: Array<{ value: string; label: string }>
    employerTypes?: Array<{ value: string; label: string }>
    salarySection16Components?: Array<{ id: string | number; label: string }>
    salarySection171Components?: Array<{ // New grouped format
      group: string
      options: Array<{ id: string | number; label: string }>
    }>
    salarySection172Components?: Array<{ id: string | number; label: string }>
    salarySection173Components?: Array<{ id: string | number; label: string }>
  }

File: api/filing/models/salary_model.py
from typing import Optional
from .api_base_model import ApiBaseModel
from .employer_model import EmployerModel
from .employer_address_model import EmployerAddressModel
from .employment_period_model import EmploymentPeriodModel
from .salary_deduction_16_model import SalaryDeduction16Model
from .salary_section_171_model import SalarySection171Model
from .salary_section_172_model import SalarySection172Model
from .salary_section_173_model import SalarySection173Model


class SalaryModel(ApiBaseModel):
    employer: EmployerModel
    employer_address: Optional[EmployerAddressModel] = None
    employment_period: Optional[EmploymentPeriodModel] = None
    salary_deduction_16: Optional[SalaryDeduction16Model] = None
    salary_section_171: list[SalarySection171Model] = []
    salary_section_172: list[SalarySection172Model] = []
    salary_section_173: list[SalarySection173Model] = []
# EmployerModel